[SDK-1955] Remove references to Auth0.OpenIdConnectSigningKeyResolver (#9465)
auth0/docs: articles/quickstart/backend/webapi-owin/01-authorization.md

### Install dependencies

To use Auth0 Access Tokens with ASP.NET you will use the OWIN JWT Middleware which is available in the `Microsoft.Owin.Security.Jwt` NuGet package.

```bash
Install-Package Microsoft.Owin.Security.Jwt
```

### Verifying the token signature

> As the OWIN JWT middleware doesn't use Open ID Connect Discovery by default, you will need to provide a custom `IssuerSigningKeyResolver`. Such a custom resolver was previously published as part of the `Auth0.OpenIdConnectSigningKeyResolver` package through Nuget. As this package is not available anymore because it was only a thin wrapper, you will need to provide this yourself.

Reviewer: I would place the second part of this as a side note, as it doesn't relate directly to the task, but does provide useful context. I would also add something task related to intro the code snippet to the reader.

Also modified the reasoning behind the package not being available (it was not available simply because it was a thin wrapper), and linked to the security notice.

Something like:

```suggestion
As the OWIN JWT middleware doesn't use Open ID Connect Discovery by default, you will need to provide a custom `IssuerSigningKeyResolver`. To do this, add the following to the `Support/OpenIdConnectSigningKeyResolver.cs` file:

:::note
Such a custom resolver was previously published as part of the `Auth0.OpenIdConnectSigningKeyResolver` package through Nuget. As [this package is not available anymore](https://github.com/auth0/auth0-aspnet-owin/blob/master/SECURITY-NOTICE.md), you will need to provide this yourself.
:::
```

```javascript

> // Support/OpenIdConnectSigningKeyResolver.cs

Reviewer: If you take the above suggestion with the intro for creating a new file, feel free to remove this line (48) - no need to repeat the filename.

public class OpenIdConnectSigningKeyResolver
{
    private readonly OpenIdConnectConfiguration openIdConfig;

    public OpenIdConnectSigningKeyResolver(string authority)
    {
        var cm = new ConfigurationManager<OpenIdConnectConfiguration>($"{authority.TrimEnd('/')}/.well-known/openid-configuration", new OpenIdConnectConfigurationRetriever());
        openIdConfig = AsyncHelper.RunSync(async () => await cm.GetConfigurationAsync());
    }

    public SecurityKey[] GetSigningKey(string kid)
    {
        return new[] { openIdConfig.JsonWebKeySet.GetSigningKeys().FirstOrDefault(t => t.KeyId == kid) };
    }
}
```

> The `OpenIdConnectSigningKeyResolver` will automatically download the JSON Web Key Set which was used to sign the RS256 tokens by interrogating the OpenID Connect Configuration endpoint (at `/.well-known/openid-configuration`). You can then use it subsequently to resolve the Issuer Signing Key, as will be demonstrated in the JWT registration code below.

Reviewer: "interrogating" sounds pretty heavy. How about:

```suggestion
The `OpenIdConnectSigningKeyResolver` will automatically download the JSON Web Key Set used to sign the RS256 tokens from the OpenID Connect Configuration endpoint (at `/.well-known/openid-configuration`). You can then use it subsequently to resolve the Issuer Signing Key, as will be demonstrated in the JWT registration code below.
```

Author: That's a fair point. This is a paragraph that I didn't touch other than move it. But I like the improvement, will update it!
